### PR TITLE
Add checksum for py-arrow 1.2.1 and 1.2.2

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-arrow/package.py [hunk 1]
```diff
@@ -27,3 +29,4 @@ class PyArrow(PythonPackage):
         "py-backports-functools-lru-cache@1.2.1:", type=("build", "run"), when="^python@2.7:2.8"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arrow/package.py:30]: Python 3.6+ required now, python-dateutil 2.7+ required now.

File: var/spack/repos/builtin/packages/py-arrow/package.py [hunk 2]
```diff
@@ -27,3 +29,4 @@ class PyArrow(PythonPackage):
...
     )
     depends_on("py-python-dateutil", type=("build", "run"))
+    depends_on("py-typing-extensions", type=("build", "run"), when="@1.2.1: ^python@:3.8")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arrow/package.py]:
```suggestion
    depends_on("py-typing-extensions", type=("build", "run"), when="@1.2.1: ^python@:3.7")
```
Spack version ranges are inclusive

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-arrow/package.py [hunk 3]
```diff
@@ -17,13 +17,18 @@ class PyArrow(PythonPackage):
...
 
+    version("1.2.2", sha256="05caf1fd3d9a11a1135b2b6f09887421153b94558e5ef4d090b567b47173ac2b")
+    version("1.2.1", sha256="c2dde3c382d9f7e6922ce636bf0b318a7a853df40ecb383b29192e6c5cc82840")
     version("0.16.0", sha256="92aac856ea5175c804f7ccb96aca4d714d936f1c867ba59d747a8096ec30e90a")
     version("0.14.7", sha256="67f8be7c0cf420424bc62d8d7dc40b44e4bb2f7b515f9cc2954fb36e35797656")
     version("0.14.1", sha256="2d30837085011ef0b90ff75aa0a28f5c7d063e96b7e76b6cbc7e690310256685")
 
-    depends_on("python@2.7:2.8,3.5:", type=("build", "run"))
+    depends_on("python@3.6:", type=("build", "run"), when="@:0.16.0")
+    depends_on("python@2.7:2.8,3.5:", type=("build", "run"), when="@1.2.1:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arrow/package.py]: I think these `when` are reversed

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-arrow/package.py]: Whoops

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**